translator/writer/prim_writer.cpp

```diff
@@ -652,4 +658,55 @@ void UsdArnoldPrimWriter::writeMatrix(UsdGeomXformable &xformable, const AtNode
             m[i][j] = matrix[i][j];
 
     xformOp.Set(GfMatrix4d(m));
+}
+
+void UsdArnoldPrimWriter::writeMaterialBinding(const AtNode *node, UsdPrim &prim, UsdArnoldWriter &writer)
+{
+    AtNode *shader = (AtNode*) AiNodeGetPtr(node, "shader");
+
+    static const AtString polymesh_str("polymesh");
+    AtNode *displacement = (AiNodeIs(node, polymesh_str)) ?
+        (AtNode*) AiNodeGetPtr(node, "disp_map"): nullptr;
+
+    if (shader == nullptr && displacement == nullptr)
+        return; // nothing to export
+
+    std::string shaderName = (shader) ? UsdArnoldPrimWriter::getArnoldNodeName(shader) : "";
+    std::string dispName = (displacement) ? UsdArnoldPrimWriter::getArnoldNodeName(displacement) : "";
+    
+    // The material node doesn't exist in Arnold, but is required in USD, 
+    // let's create one based on the name of the shader plus the name of 
+    // the eventual displacement. This way we'll have a unique material in USD
+    // per combination of surface shader + displacement instead of duplicating it
+    // for every geometry.
+    std::string materialName = "/materials";
+    materialName += shaderName;
```

Comment:
The returned `shaderName` and `dispName` will both start with a `/` creating a child in the `/materials` scope. I think that's good because it allows us to organize the materials nicely.

However, it'll create a weird hierarchy inside the materials, especially when displacement shaders are used. For example, our scene has two shaders, `ShaderA` (surface) and `ShaderB` (displacement). If we have two meshes, `A`, to which only `ShaderA` is assigned, and `B` to which both `ShaderA` and `ShaderB` is assigned, we'll end up with a layout like this:

```
def "materials"
{
    def Material "ShaderA"
    {
        # ...
    }

   def "ShaderA_disp_"
   {
        def Material "ShaderB"
        {
             # ...
        }
   }
}
```

Is this intentional?